### ISSUE TITLE
SSIM as an Objective?

### ISSUE THREAD

**Issue by Sentient07** [Lasagne/Lasagne#881]
Recently, I have been working with DAEs and found the problem of over-smoothness in denoised images(as I was using MSE as loss function). There have been recent works[1] on replacing MSE with SSIM to overcome this problem. Would be great to have SSIM as objective in Lasagne. Related work in Keras [2]

[1] https://arxiv.org/pdf/1602.02644.pdf 
[2] https://github.com/fchollet/keras/issues/4292
